src/compile/data.ts

```diff
@@ -564,7 +667,16 @@ export namespace summary {
     return [];
   }
 
-  export function assemble(component: DataComponent): VgData[] {
+  export function parseLayer(model: LayerModel): SummaryComponent[] {
+    // we create new data sources for each summary in a layer
+    return [];
+  }
+
+  /**
+   * Assemble the summary. Needs a rename function because we cannot guarantee that the
+   * parent data before the children data.
+   */
+  export function assemble(component: DataComponent, rename: (name: string) => string): VgData[] {
```

Comment:
That's not what we discussed. We discussed that we should not set the names in parse ;-P